react/react2/week1/todo-list-app/src/App.js

```diff
@@ -39,124 +43,28 @@ const App = () => {
       setIsEmpty(true)
     }
   }, [query, url])
+
+  const contextValue = {
+    inputHandler: inputHandler,
+    users: users,
+    title: 'GitHub user searcher',
+  }
+
   return (
-    <>
-      <Header title={'GitHub user searcher'} />
+    <UserContext.Provider value={contextValue}>
       <div className='container'>
-        <Input inputHandler={inputHandler} />
-        {isLoading && <ReaquestStatus status={'LOADING ...'} />}
-        {isError && <ReaquestStatus status={'ERROR ... 404. Bad request'} />}
+        <Header />
+        <Input />
+        {isLoading && <RequestStatus status={'LOADING ...'} />}
```

Comment:
why are you not handling the same as "usersList" component, would look much cleaner. In general loading, error, users could be imagined as switch function, where only one outcome is possible at a time